app/commands/v2/publish.rb

```diff
@@ -163,6 +167,40 @@ def set_public_updated_at
         end
       end
 
+      def set_temporary_major_published_at
```

Comment:
Up to you on this one, but I would maybe reduce this into one method, feels a bit like overkill to me at the moment.

```ruby
def set_temporary_published_at_dates
  edition.update_attributes!(temporary_major_published_at: default_datetime) if update_type == "major"
  edition.update_attributes!(temporary_minor_published_at: default_datetime) if update_type == "minor"
end
```